presto-expressions/src/main/java/com/facebook/presto/expressions/translator/TranslatorAnnotationParser.java

```diff
@@ -124,43 +126,80 @@ private static List<ScalarHeaderAndMethods> findScalarsFromSetClass(Class<?> cla
     private static Map<FunctionMetadata, MethodHandle> scalarToFunctionMetadata(ScalarHeaderAndMethods scalar)
     {
         ScalarTranslationHeader header = scalar.getHeader();
-
         ImmutableMap.Builder<FunctionMetadata, MethodHandle> metadataBuilder = ImmutableMap.builder();
         for (Method method : scalar.getMethods()) {
-            FunctionMetadata metadata = methodToFunctionMetadata(header, method);
-            metadataBuilder.put(removeTypeParameters(metadata), getMethodHandle(method));
+            List<FunctionMetadata> allMetadata = methodToFunctionMetadata(header, method);
+            for (FunctionMetadata metadata : allMetadata) {
+                metadataBuilder.put(removeTypeParameters(metadata), getMethodHandle(method));
+            }
         }
-
         return metadataBuilder.build();
     }
 
-    private static FunctionMetadata methodToFunctionMetadata(ScalarTranslationHeader header, Method method)
+    private static List<FunctionMetadata> methodToFunctionMetadata(ScalarTranslationHeader header, Method method)
     {
         requireNonNull(header, "header is null");
 
-        // return type
         SqlType annotatedReturnType = method.getAnnotation(SqlType.class);
-        checkArgument(annotatedReturnType != null, format("Method [%s] is missing @SqlType annotation", method));
-        TypeSignature returnType = parseTypeSignature(annotatedReturnType.value(), ImmutableSet.of());
-
-        // argument type
-        ImmutableList.Builder<TypeSignature> argumentTypes = new ImmutableList.Builder<>();
-        for (Parameter parameter : method.getParameters()) {
-            Annotation[] annotations = parameter.getAnnotations();
-
-            SqlType type = Stream.of(annotations)
-                    .filter(SqlType.class::isInstance)
-                    .map(SqlType.class::cast)
-                    .findFirst()
-                    .orElseThrow(() -> new IllegalArgumentException(format("Method [%s] is missing @SqlType annotation for parameter", method)));
-            TypeSignature typeSignature = parseTypeSignature(type.value(), ImmutableSet.of());
-            argumentTypes.add(typeSignature);
-        }
+        SupportedSignatures supportedSignatures = method.getAnnotation(SupportedSignatures.class);
```

Comment:
I just saw this if block, I think migrating the Clickhouse OperatorTranslators right away makes sense to avoid the need for this if block